### PR TITLE
[python] - cpython - prev. version fallback - fix

### PR DESCRIPTION
**Pull request by gauravsaini04**
**Feature name**:
 
 * Python
 
 **Description**:
 
 This PR introduces the following functionality:
 
 * When the latest tag for `cpython` doesn't have a source binary released yet, then it will continue installing the previous working version for which a source binary will be available
 
 _Changelog_:
 
 * Updated install.sh
 * Updated tests to validate that when a fake `cpython` version is tried to be downloaded with curl command thus representing trying to access a source binary which is not deployed yet, then will pick up the binary for the tag which will eventually work successfully.
 * Updated patch version in `devcontainer-feature.json`
 
 **Checklist**:
 
 * [x]   Checked that applied changes work as expected

### CODE REVIEW

**Review by samruddhikhandale — APPROVED** [review]
Looks great, one minor comment

**Review by samruddhikhandale — DISMISSED** [review]
Looks great, one minor comment

File: src/python/install.sh [hunk 1]
```diff
@@ -396,23 +454,23 @@ install_from_source() {
...
-    local tgz_filename="Python-${VERSION}.tgz"
-    local tgz_url="https://www.python.org/ftp/python/${VERSION}/${tgz_filename}"
-    echo "Downloading ${tgz_filename}..."
-    curl -sSL -o "/tmp/python-src/${tgz_filename}" "${tgz_url}"
-
+    install_cpython "${VERSION}"
+    if [ -f "/tmp/python-src/${cpython_tgz_filename}" ]; then 
+        if grep -q "404 Not Found" "/tmp/python-src/${cpython_tgz_filename}"; then
+            # Use grep to search for "404 Not Found" in the file
+            echo "\"404 Not Found\" found in /tmp/python-src/${cpython_tgz_filename}. Not able to create source binary"
```
Comment (samruddhikhandale) [inline src/python/install.sh:461]: Looks like this was used for debugging, remove?

File: src/python/devcontainer-feature.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "id": "python",
-  "version": "1.4.1",
+  "version": "1.4.2",
```
Comment (samruddhikhandale) [inline src/python/devcontainer-feature.json:3]: Looks like tests are failing with `configure: error: Invalid --with-openssl value`, can you help fix them?

File: src/python/install.sh (hunk 1 above)
Comment (gauravsaini04) [reply src/python/install.sh]: removed 👍

File: src/python/devcontainer-feature.json (hunk 2 above)
Comment (gauravsaini04) [reply src/python/devcontainer-feature.json:3]: done. not failing now 👍

**Review by samruddhikhandale — APPROVED** [review]
Thank you 👏

**Pull request merged**